runtime/win32.c

```diff
@@ -1272,3 +1271,32 @@ value caml_win32_xdg_defaults(void)
 
   CAMLreturn(result);
 }
+
+static DWORD (WINAPI *get_temp_path)(DWORD, LPWSTR);
+static INIT_ONCE get_temp_path_init_once = INIT_ONCE_STATIC_INIT;
+static BOOL CALLBACK get_temp_path_init_function(PINIT_ONCE InitOnce,
+                                                 PVOID Parameter,
+                                                 PVOID *lpContext)
+{
+  get_temp_path = (DWORD (WINAPI *)(DWORD, LPWSTR))
+    GetProcAddress(GetModuleHandle(L"KERNEL32.DLL"), "GetTempPath2W");
+  if (!get_temp_path)
+    get_temp_path = GetTempPath;
+  return TRUE;
+}
+
+value caml_win32_get_temp_path(void)
+{
+  CAMLparam0();
+  CAMLlocal1(path);
+  wchar_t buf[MAX_PATH+1];
+
+  InitOnceExecuteOnce(&get_temp_path_init_once, get_temp_path_init_function,
+                      NULL, NULL);
```

Comment:
No error checking?

Reply:
The [documentation of [`InitOnceExecuteOnce`]() reads:

> If the callback function returns `TRUE` to indicate success, `InitOnceExecuteOnce` will return `TRUE` back to all callers at once. If, however, the callback returns `FALSE` to indicate failure, `InitOnceExecuteOnce` will return `FALSE` to only the single thread which executed the callback function. At this same time one of the remaining blocked threads will unblock and execute `InitFn` once again. Thus, in a scenario where `InitFn` can fail intermittently and retries are desired, all threads should continue calling `InitOnceExecuteOnce` until `TRUE` is returned.

It doesn't seem that `InitOnceExecuteOnce` can error on its own.

The callback function is:
```c
static BOOL CALLBACK get_temp_path_init_function(PINIT_ONCE InitOnce,
                                                 PVOID Parameter,
                                                 PVOID *lpContext)
{
  get_temp_path = (DWORD (WINAPI *)(DWORD, LPWSTR))
    GetProcAddress(GetModuleHandle(L"KERNEL32.DLL"), "GetTempPath2W");
  if (!get_temp_path)
    get_temp_path = GetTempPath;
  return TRUE;
}
```
I could be mistaken, but I don't think there's a world in which these may fail.

Reply:
Thanks for the explanation!